### PR TITLE
Fix the flow operations, due to the addition of the mirror traffic feature

### PR DESCRIPTION
**Pull request by dpoltavets**
Related to: #3849

**MERGE NOTE: DO NOT MERGE BEFORE MERGING #4238**

### CODE REVIEW

File: src-java/base-topology/base-messaging/src/main/java/org/openkilda/messaging/model/MirrorPathStatusDto.java [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+import java.io.Serializable;
+
+@Data
+@NoArgsConstructor
+@AllArgsConstructor
+@JsonInclude(JsonInclude.Include.NON_NULL)
+@JsonNaming(SnakeCaseStrategy.class)
+public class MirrorPathStatusDto implements Serializable {
+    String mirrorPathId;
```
Comment (surabujin) [inline src-java/base-topology/base-messaging/src/main/java/org/openkilda/messaging/model/MirrorPathStatusDto.java:33]: is this our internal pathId? As for me we should define mirror point by switch+mirror_id or by switch+direction+mirror_id. Just mIrror-id should be enought too, but it forces client to have internal database and map mirro-id into switch+direction on it's side.

File: src-java/base-topology/base-storm-topology/src/main/java/org/openkilda/wfm/share/mappers/FlowMapper.java [hunk 2]
```diff
@@ -69,14 +73,18 @@
...
     public abstract FlowDto map(Flow flow);
 
     /**
      * Convert {@link Flow} to {@link FlowDto} with diverse flow ids.
      */
-    public FlowDto map(Flow flow, Set<String> diverseWith) {
+    public FlowDto map(Flow flow, Set<String> diverseWith, List<FlowMirrorPath> flowMirrorPaths) {
         FlowDto flowDto = map(flow);
         flowDto.setDiverseWith(diverseWith);
+        flowDto.setMirrorPathStatuses(flowMirrorPaths.stream()
```
Comment (surabujin) [inline src-java/base-topology/base-storm-topology/src/main/java/org/openkilda/wfm/share/mappers/FlowMapper.java:85]: It should be possible to do via mapstruct.

**Review by surabujin — CHANGES_REQUESTED** [review]
There is no FSM diagramm.

File: src-java/northbound-service/northbound-api/src/main/java/org/openkilda/northbound/dto/v2/flows/MirrorPathStatus.java [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+import lombok.AllArgsConstructor;
+import lombok.Builder;
+import lombok.Data;
+
+@Data
+@Builder
+@AllArgsConstructor
+@JsonNaming(value = SnakeCaseStrategy.class)
+public class MirrorPathStatus {
+    String mirrorPathId;
```
Comment (rtretyak) [inline src-java/northbound-service/northbound-api/src/main/java/org/openkilda/northbound/dto/v2/flows/MirrorPathStatus.java:29]: mirrorPointId ?

File: src-java/base-topology/base-storm-topology/src/main/java/org/openkilda/wfm/share/mappers/FlowMapper.java (hunk 2 above)
Comment (dpoltavets) [reply src-java/base-topology/base-storm-topology/src/main/java/org/openkilda/wfm/share/mappers/FlowMapper.java]: Now mapstruct is used

File: src-java/base-topology/base-messaging/src/main/java/org/openkilda/messaging/model/MirrorPathStatusDto.java (hunk 1 above)
Comment (dpoltavets) [reply src-java/base-topology/base-messaging/src/main/java/org/openkilda/messaging/model/MirrorPathStatusDto.java]: Yes, this is our internal pathId for mirror point.

**Comment by dpoltavets** [comment]
FSM diagrams do not need to be updated, since the states and transitions between them have not been changed.

**Review by niksv — APPROVED** [review]

**Review by surabujin — APPROVED** [review]

**Pull request merged**